### ISSUE TITLE
How evaluation is done from captions of Validation Set

### ISSUE THREAD

**Issue by deepikachettri29** [facebookresearch/stable_signature#36]
Hello author,
Can you please share how evaluation is done on generation om prompts from validation set of COCO dataset(Appendix A.3) in Text-to-Image. It would be helpful if you could share how evaluation is done while doing inpainting,image edition and super resolution. I'm trying to get similar results as yours.

**Comment by pierrefdz** [comment]
Here are the prompts I used for COCO
https://huggingface.co/datasets/pierrefdz/stable-signature/blob/main/coco-prompts-5k.txt

For inpainting, we used DiffEdit, you can find an implementation here:
https://github.com/Xiang-cd/DiffEdit-stable-diffusion

For super resolution, I used
https://huggingface.co/datasets/pierrefdz/stable-signature/blob/main/super-res.py
, adapted from the superres file of https://github.com/Stability-AI/stablediffusion/blob/main/scripts/streamlit/superresolution.py
(You need to run this from the stablediffusion repository)

**Comment by deepikachettri29** [comment]
Thank you for quick reply . Could you please share me the code too for the first text-to-image generation.

**Comment by pierrefdz** [comment]
https://github.com/Stability-AI/stablediffusion/blob/main/scripts/txt2img.py

You just need to reload the decoder's weights before generation

```
    # Parameter None for clutil sweep
    print(f'reload decoder weights {opt.ldm_decoder_ckpt}...')
    if opt.ldm_decoder_ckpt is not None and opt.ldm_decoder_ckpt.lower() == "none":
        opt.ldm_decoder_ckpt = None
    if opt.ldm_decoder_ckpt is not None:
        state_dict = torch.load(opt.ldm_decoder_ckpt)['ldm_decoder']
        # state_dict = torch.load(opt.ldm_decoder_ckpt)['state_dict']
        # state_dict = {k.replace('first_stage_model.', ''): v for k, v in state_dict.items() if 'decoder' in k or 'post_quant_conv' in k}
        msg = model.first_stage_model.load_state_dict(state_dict, strict=False)
        model.eval()
        print(msg)

````

**Comment by deepikachettri29** [comment]
Thank you for this too. Also can it be done with the diffusers library too where it can generate 512x512 images?

**Comment by deepikachettri29** [comment]
Also did you run the code again for getting non watermarked images with --ldm_decoder_ckpt none and then calculated fid score?

**Comment by pierrefdz** [comment]
> Thank you for this too. Also can it be done with the diffusers library too where it can generate 512x512 images?

I think so but I haven't tried to be honest. The stablediffusion codebase can generate images at 512x512 as well.

> Also did you run the code again for getting non watermarked images with --ldm_decoder_ckpt none and then calculated fid score?

Yes exactly. In the paper we report the FID on the watermarked generated images, as well as the difference compared to without watermark. I don't remember exactly the parameters of the FID though (it's very dependant on the total number of images).
